### PR TITLE
Run boefje, normalizer and bit from cli

### PR DESCRIPTION
**Pull request by dekkers**
### Changes
This adds four scripts so we can run a boefje, normalizer and bit from the command line and easily see the contents a raw file.

This adds a new bytes API call to get the raw file contents by id that is used by show_raw. 

When debugging things I noticed the `validation_exception_handler` in bytes was broken so this is also fixed in this PR.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
nice work. untested, but looks good.

**Review by ammar92 — COMMENTED** [review]
Nice one, reminds me of the old days where we had similar tools for running and testing boefjes and normalizers.

One remark: I do think we should make the user responsible for setting the correct paths, or document that this tool should be run from the module directory and not the tools directory. This may be a bit opinionated, but this way is how I'm used to run stuff like this. (and to get rid of the `E402`)

**Comment by underdarknl** [comment]
Maybe this could also use a list command for each?

**Comment by Darwinkel** [comment]
I got the tools to work from inside the containers, but we should really provide some proper documentation regarding their use.

As @underdarknl says, having some list-like commands would be great, for example to show which BIT, BOEFJE, RAW, and NORMALIZER id's are available. Would be nice if people didn't have to dig through the source code and/or the database :)

**Comment by dekkers** [comment]
> One remark: I do think we should make the user responsible for setting the correct paths, or document that this tool should be run from the module directory and not the tools directory. This may be a bit opinionated, but this way is how I'm used to run stuff like this. (and to get rid of the `E402`)

I disagree. It doesn't give any benefit and only makes the tools harder to use. Ease of use is more important than not having ` sys.path.insert` line in the code. Running the tools from the module directory doesn't work btw, because the current work directory isn't in the python path.

**Review by ammar92 — DISMISSED** [review]
Will test this

File: docs/source/developer_documentation/boefjes.md [hunk 1]
```diff
@@ -182,12 +182,38 @@ The job file for a DNS scan might look like this:
...
+
+Both `run_boefje.py` and `run_normalizer.py` support the `--pdb` option to enter
+the standard Python Debugger when an exceptions happens or breakpoint is
+triggered.
+
+If you are using the standard docker compose developer setup, you can use
+`docker compose exec` to execute the commands in the container:
+
+```shell
+$ docker compose exec boefje ./tools/run_boefje.py ORGANIZATION_CODE BOEFJE_ID INPUT_OOI
```
Comment (Darwinkel) [inline docs/source/developer_documentation/boefjes.md:215]: Maybe note that both the `boefje` as well as the `normalizer` containers can be used for this?

**Review by Darwinkel — COMMENTED** [review]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
- `docker compose exec boefje ./tools/run_boefje.py henk dns-records "Hostname|internet|mispo.es"` works
- `docker compose exec boefje ./tools/show_raw.py 794986d7-cf39-4a2c-8bdf-17ae58f361ea --json` works
- When using the wrong parameters, this is reasonably obvious from the errors given.

### What doesn't work:
- `docker compose exec octopoes_api ./tools/run_bit.py henk missing-dmarc "Hostname|internet|domaindiscount24.net"` results in the below error:

```
DEBUG:pika.callback:Processing use of oneshot callback
DEBUG:pika.callback:0 registered uses left
DEBUG:pika.callback:Comparing {'queue': 'henk__scan_profile_mutations'} to {'queue': 'henk__scan_profile_mutations'}
DEBUG:pika.callback:Removing callback #0: {'callback': <bound method _CallbackResult.set_value_once of <pika.adapters.blocking_connection._CallbackResult object at 0x7f597511c680>>, 'one_shot': True, 'only': None, 'arguments': {'queue': 'henk__scan_profile_mutations'}, 'calls': 0}
DEBUG:pika.callback:Calling <bound method Channel._on_synchronous_complete of <Channel number=1 OPEN conn=<SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f5974523450> params=<URLParameters host=rabbitmq port=5672 virtual_host=kat ssl=False>>>> for "1:Queue.DeclareOk"
DEBUG:pika.channel:0 blocked frames
DEBUG:pika.callback:Calling <bound method _CallbackResult.set_value_once of <pika.adapters.blocking_connection._CallbackResult object at 0x7f597511c680>> for "1:Queue.DeclareOk"
DEBUG:pika.adapters.utils.io_services_utils:Recv would block on <socket.socket fd=6, family=2, type=1, proto=6, laddr=('172.21.0.12', 38986), raddr=('172.21.0.6', 5672)>
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): crux:3000
DEBUG:urllib3.connectionpool:http://crux:3000 "GET /_xtdb/henk/entity?eid=Hostname%7Cinternet%7Cdomaindiscount24.net&valid-time=2023-10-17T12%3A30%3A55.115969%2B00%3A00 HTTP/1.1" 200 276
DEBUG:urllib3.connectionpool:http://crux:3000 "GET /_xtdb/henk/entity?eid=Origin%7Cinference%7Cmissing-dmarc%7CHostname%7Cinternet%7Cdomaindiscount24.net&valid-time=2023-10-17T12%3A30%3A55.115969%2B00%3A00 HTTP/1.1" 200 316
Traceback (most recent call last):
  File "/app/octopoes/./tools/run_bit.py", line 76, in <module>
    run_bit()
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/octopoes/./tools/run_bit.py", line 59, in run_bit
    path = Path.parse(f"{param_def.ooi_type.get_object_type()}.{param_def.relation_path}").reverse()
           ^^^^^^^^^^
AttributeError: type object 'Path' has no attribute 'parse'. Did you mean: 'parts'?
```

### Bug or feature?:
- Maybe it would be nice to give some concrete examples of usage in the documentation? Getting the tools to work isn't too hard, but it was a bit of trial and error to find the right parameters (feel free to copy-paste the above commands as reference examples ;) )

**Review by Darwinkel — APPROVED** [review]
Fix confirmed working. Docs look great!

**Review by underdarknl — APPROVED** [review]

**Pull request merged**